docs/sphinx/api/builtin_ops.rst

```diff
@@ -0,0 +1,255 @@
+Built-in Quantum Operations
+******************************
+
+CUDA Quantum provides a default set of built-in operations on qubits. 
+These built-in operations can be used to define custom kernels and libraries.
+Since the set of quantum intrinsic operations natively supported on a specific target 
+depends on the backends architecture, the :code:`nvq++` compiler automatically
+decomposes built-in operations into the appropriate set of intrinsic operations 
+for that target.
+
+Built-in operations that implement unitary transformations of the quantum state are templated. 
+The template argument :code:`cudaq::adj` can be used to invoke the 
+`adjoint <https://en.wikipedia.org/wiki/Conjugate_transpose>`__ transformation:
```

Comment:
I'd recommend putting the `cudaq:adj` and `cudaq::ctrl` sections (currently lines 11-81) after the gate section (aka `Unitary Operations on Qubit`). Otherwise we are referencing gates that haven't been defined yet.

Reply:
I moved the paragraphs into their own section, and added a sentence here at the top that forward references all subsequent sections.